### PR TITLE
Feature/5967 extended analytics

### PR DESCRIPTION
**Pull request by thomasridd**
### What

Added extended analytics tracking when users download a file as part of the frontend-dataset-controller
https://trello.com/c/zkeB3QbA

### How to review

Sense check
Tests pass

### Who can review

Golang frontend. !me

### CODE REVIEW

File: mapper/census.go [hunk 1]
```diff
@@ -361,6 +364,18 @@ func CreateCensusDatasetLandingPage(isEnableMultivariate bool, ctx context.Conte
 		p.DatasetLandingPage.QualityStatements[qsLen-1].CssClasses = append(p.DatasetLandingPage.QualityStatements[qsLen-1].CssClasses, "ons-u-mb-l")
 	}
 
+	if isFilterOutput {
+		analyticsJavaScript, err := getDataLayerJavaScript(getFilterAnalytics(fDims))
```
Comment (mr-nick17) [inline mapper/census.go:368]: What about if there is an error? As you're not handling it, presuming that you don't want the server to error at all? If that's the case, I'd suggest the following
```suggestion
		analyticsJavaScript := getDataLayerJavaScript(getFilterAnalytics(fDims))
```

File: mapper/census.go [hunk 2]
```diff
@@ -524,3 +539,56 @@ func generateTruncatePath(path, dimID string, q url.Values) string {
 	}
 	return truncatePath
 }
+
+func getDataLayerJavaScript(analytics map[string]string) (template.JS, error) {
+	jsonStr, err := json.Marshal(analytics)
+	if err != nil {
```
Comment (mr-nick17) [inline mapper/census.go:545]: Following on from above...
```suggestion
	jsonStr, _ := json.Marshal(analytics)
```

File: mapper/census.go [hunk 3]
```diff
@@ -524,3 +539,56 @@ func generateTruncatePath(path, dimID string, q url.Values) string {
...
+	analytics := make(map[string]string, 5)
+	var dimensionIDs []string
+	for _, filterDimension := range filterDimensions {
+		dimension := filterDimension.ModelDimension
+		if dimension.IsAreaType != nil && *dimension.IsAreaType {
+			analytics["areaType"] = dimension.ID
+			analytics["coverageCount"] = strconv.Itoa(len(dimension.Options))
+
+			if len(dimension.Options) > 0 {
+				if len(dimension.Options) <= 4 {
```
Comment (mr-nick17) [inline mapper/census.go:578]: The int `4` is a bit of a magic number; I had to read the a/c on the ticket to understand why. Please either add a comment or an appropriately named variable

**Review by mr-nick17 — COMMENTED** [review]
A comment/suggestion about error handling (or not in this case). If you decide not to do anything with the error, can you add a small note explaining why.

**Comment by thomasridd** [comment]
I've decided not to do anything about the error - it was bubbling up from json.Marshal which since it is only marshalling a simple map won't fail

**Review by mr-nick17 — APPROVED** [review]

**Pull request merged**